### PR TITLE
:bug: Filter XSS injection

### PR DESCRIPTION
**Pull request by JaderH**
Fixing issue #30 

image

### CODE REVIEW

**Comment by antonkomarev** [comment]
Thank you for the contribution @JaderH! Will be merged after I find a time to test it.

**Comment by antonkomarev** [comment]
Why do we need ENT_NOQUOTES flag to be set?
It looks like simple `$input = htmlspecialchars($input);` is enough.

File: public/file-repository.php [hunk 1]
```diff
@@ -38,6 +38,10 @@
         $storagePath = $_ENV['FILE_STORAGE_PATH'];
     }
 
+    array_walk_recursive($_GET, function (&$input) {
```
Comment (antonkomarev) [inline public/file-repository.php:41]: `array_walk_recursive` may be safely replaced with `array_walk`. We do not support multidimensional GET params

**Comment by JaderH** [comment]
> Why do we need ENT_NOQUOTES flag to be set?
> It looks like simple `$input = htmlspecialchars($input);` is enough.

`ENT_NOQUOTES` is to encode incorrect single or double quotes. If single or double quotes are encoded, the x attribute of svg will increase

`ENT_NOQUOTES` Unused examples
image

`ENT_NOQUOTES` Examples are used
image

File: public/file-repository.php (hunk 1 above)
Comment (JaderH) [reply public/file-repository.php:41]: Multi-dimensional GET parameters are not supported, but we cannot prevent users from transmitting. When multi-dimensional parameters appear, PHP Fatal error will appear when using array_walk

```
[error] 31#31: *206 FastCGI sent in stderr: "PHP message: PHP Fatal error:  Uncaught TypeError: htmlspecialchars() expects parameter 1 to be string, array given in /webser/www/github-profile-views-counter/public/file-repository.php:42
Stack trace:
#0 /webser/www/github-profile-views-counter/public/file-repository.php(42): htmlspecialchars()
#1 [internal function]: {closure}()
#2 /webser/www/github-profile-views-counter/public/file-repository.php(43): array_walk()
#3 /webser/www/github-profile-views-counter/public/index.php(14): require('/webser/www/git...')
```

**Comment by antonkomarev** [comment]
Thanks for providing some more details!

**Pull request merged**